client/components/ui/sidebar/Sidebar.vue

```diff
@@ -0,0 +1,161 @@
+        </PButton>
+      </div>
+    </div>
+    <div class="items flex flex-col justify-between" style="flex-grow: 1;">
+      <div class="flex flex-col justify-start gap-2 p-4">
+        <SidebarButton
+          icon="i-tabler-file" :label="t('pages.files.page_name')" :open="sidebarOpen"
+          :active="route => route.startsWith('/files')" @click="navigateTo('/files')"
+        />
+        <SidebarButton
+          icon="i-tabler-star" :label="t('pages.favorites.page_name')" :open="sidebarOpen"
+          :active="route => route.startsWith('/favorites')" @click="navigateTo('/favorites')"
+        />
+        <SidebarButton
+          icon="i-tabler-upload" :label="t('pages.upload.page_name')" :open="sidebarOpen"
+          :active="route => route.startsWith('/upload')" @click="navigateTo('/upload')"
+        />
+        <SidebarButton
+          icon="i-tabler-timeline" :label="t('pages.analytics.page_name')" :open="sidebarOpen"
+          :active="route => route.startsWith('/analytics')" @click="navigateTo('/analytics')"
+        />
+        <SidebarButton
+          icon="i-tabler-settings" :label="t('pages.configuration.page_name')" :open="sidebarOpen"
+          :active="route => route.startsWith('/configuration')" @click="navigateTo('/configuration')"
+        />
+      </div>
+      <div class="flex flex-col p-4 gap-2">
+        <SidebarButton icon="i-tabler-help-circle" :label="t('sidebar.docs')" :open="sidebarOpen" />
+        <SidebarButton :icon="themeIcon" :label="themeName" :open="sidebarOpen" @click="toggle()" />
+        <div class="sep" />
+        <SidebarButton icon="i-tabler-user-circle" label="{Username goes here}" :open="sidebarOpen" />
+      </div>
+    </div>
+  </aside>
+</template>
+
+<style scoped lang="scss">
+.sidebar {
+  width: 5em;
+  min-width: 5em;
+  height: 100%;
+  transition: min-width 0.3s ease, width 0.3s ease;
+  overflow: hidden;
+  background-color: var(--primary-color);
+  color: var(--primary-color-text);
+
+  &.expanded {
+    width: 19em;
+    min-width: 19em;
+  }
+
+  > .header {
+    display: flex;
+    padding: 1rem;
+    background-color: var(--primary-600);
+
+    .p-button {
+      background-color: var(--primary-600);
+
+      &:hover {
+        background-color: var(--primary-500);
+      }
+
+      &:active {
+        background-color: var(--primary-700);
+      }
+    }
+  }
+
+  > .items {
+    .p-button:hover {
+      background-color: var(--primary-400);
+    }
+
+    .p-button:active {
+      background-color: var(--primary-600);
+    }
+
+    .p-button.active {
+      background-color: var(--primary-700);
+    }
+  }
+
+  .p-button {
+    padding: .75rem;
+    transition: background-color .2s, color .2s, border-color .2s, box-shadow .2s, padding 0.3s ease;
+
+    &.open {
+      padding: .75rem 1.25rem;
+    }
+  }
+
+  .p-button, .p-button:active, .p-button:hover {
+    border: none !important;
+    outline: none !important;
+    box-shadow: none !important;
+  }
```

Comment:
```suggestion
  .p-button, .p-button:active, .p-button:hover {
    border: none;
    outline: none;
    box-shadow: none;
  }
```

seems like it works fine like that now that you're using compound selectors (thanks to scss)